solidity/test/modules/multisigModule.test.ts

```diff
@@ -0,0 +1,345 @@
+      ).to.be.revertedWith('!enrolled');
+    });
+
+    it('reverts when called by a non-owner', async () => {
+      await expect(
+        multisigModule
+          .connect(nonOwner)
+          .unenrollValidator(ORIGIN_DOMAIN, validators[0].address),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('#setThreshold', () => {
+    beforeEach(async () => {
+      // Have 2 validators to allow us to have more than 1 valid
+      // quorum threshold
+      await multisigModule.enrollValidator(
+        ORIGIN_DOMAIN,
+        validators[0].address,
+      );
+      await multisigModule.enrollValidator(
+        ORIGIN_DOMAIN,
+        validators[1].address,
+      );
+    });
+
+    it('sets the quorum threshold', async () => {
+      await multisigModule.setThreshold(ORIGIN_DOMAIN, 2);
+
+      expect(await multisigModule.threshold(ORIGIN_DOMAIN)).to.equal(2);
+    });
+
+    it('emits the SetThreshold event', async () => {
+      const expectedCommitment = getCommitment(2, [
+        validators[0].address,
+        validators[1].address,
+      ]);
+      expect(await multisigModule.setThreshold(ORIGIN_DOMAIN, 2))
+        .to.emit(multisigModule, 'ThresholdSet')
+        .withArgs(ORIGIN_DOMAIN, 2, expectedCommitment);
+    });
+
+    it('reverts if the new quorum threshold is zero', async () => {
+      await expect(
+        multisigModule.setThreshold(ORIGIN_DOMAIN, 0),
+      ).to.be.revertedWith('!range');
+    });
+
+    it('reverts if the new quorum threshold is greater than the validator set size', async () => {
+      await expect(
+        multisigModule.setThreshold(ORIGIN_DOMAIN, 3),
+      ).to.be.revertedWith('!range');
+    });
+
+    it('reverts when called by a non-owner', async () => {
+      await expect(
+        multisigModule.connect(nonOwner).setThreshold(ORIGIN_DOMAIN, 2),
+      ).to.be.revertedWith('Ownable: caller is not the owner');
+    });
+  });
+
+  describe('#validators', () => {
+    beforeEach(async () => {
+      // Must be done sequentially so gas estimation is correct.
+      for (const v of validators) {
+        await multisigModule.enrollValidator(ORIGIN_DOMAIN, v.address);
+      }
+    });
+
+    it('returns the validators', async () => {
+      expect(await multisigModule.validators(ORIGIN_DOMAIN)).to.deep.equal(
+        validators.map((v) => v.address),
+      );
+    });
+  });
+
+  describe('#validatorCount', () => {
+    beforeEach(async () => {
+      // Must be done sequentially so gas estimation is correct.
+      for (const v of validators) {
+        await multisigModule.enrollValidator(ORIGIN_DOMAIN, v.address);
+      }
+    });
+
+    it('returns the number of validators enrolled in the validator set', async () => {
+      expect(await multisigModule.validatorCount(ORIGIN_DOMAIN)).to.equal(
+        validators.length,
+      );
+    });
+  });
+
+  describe('#verify', () => {
```

Comment:
can we add a test for a situation where the threshold is less than the validator set size and the signatures provided aren't from the first `threshold` validators in the EnumerableSet? Hoping to cover the case that I think I noticed in `_verifyValidatorSignatures`